### PR TITLE
Cluster pair should work with auth enabled Portworx

### PR DESCRIPTION
**Pull request by taaraora**
**What type of PR is this?**
> Uncomment only one and also add the corresponding label in the PR:

bug
>feature
>cleanup
>api-change
>design
>documentation
>failing-test
>unit-test
>integration-test

**What this PR does / why we need it**:

**Does this PR change a user-facing CRD or CLI?**:

**Is a release note needed?**:

```release-note

```

**Does this change need to be cherry-picked to a release branch?**:

### CODE REVIEW

File: drivers/volume/portworx/portworx.go [hunk 1]
```diff
@@ -243,10 +242,32 @@ func (p *portworx) getClusterDomainClient() (api.OpenStorageClusterDomainsClient
 	return api.NewOpenStorageClusterDomainsClient(conn), nil
 }
 
+func (p *portworx) getClusterManagerClient() (cluster.Cluster, error) {
```
Comment (lpabon) [inline drivers/volume/portworx/portworx.go:245]: Add comments

File: drivers/volume/portworx/portworx.go [hunk 2]
```diff
@@ -297,6 +310,46 @@ func (p *portworx) initPortworxClients() error {
 	return err
 }
 
+func (p *portworx) tokenGenerator() (string, error) {
```
Comment (lpabon) [inline drivers/volume/portworx/portworx.go:313]: Add a comment here how it is expected that this code will be executed for those that are new to this model.

File: drivers/volume/portworx/portworx.go [hunk 3]
```diff
@@ -469,7 +528,12 @@ func (p *portworx) GetNodes() ([]*storkvolume.NodeInfo, error) {
 		}
 	}
 
-	cluster, err := p.clusterManager.Enumerate()
+	clusterManager, err:= p.getClusterManagerClient()
```
Comment (lpabon) [inline drivers/volume/portworx/portworx.go:531]: Why get cluster manager again?

Why was it removed from the object `p`?

Comment (taaraora) [reply drivers/volume/portworx/portworx.go:531]: When cluster manager had static config we could just keep it as part of `p *portworx` struct. Now when we need to populate cluster manager client with token which has relatively short lifetime (1 hour), `getClusterManagerClient()` is used as getter which returns fully configured client with fresh token injected.

File: drivers/volume/portworx/portworx.go (hunk 1 above)
Comment (taaraora) [reply drivers/volume/portworx/portworx.go:247]: done

File: drivers/volume/portworx/portworx.go (hunk 2 above)
Comment (taaraora) [reply drivers/volume/portworx/portworx.go:318]: done

**Pull request merged**